src/os/spawn.php

```diff
@@ -0,0 +1,144 @@
+<?hh
+
+namespace HH\Lib\Experimental\OS;
+use namespace HH\Lib\C;
+use namespace HH\Lib\Vec;
+use namespace \HH\Lib\_Private\_OS;
+use type HH\Lib\OS\{FileDescriptor, IsNotADirectoryException};
+use function HH\Lib\OS\{close, open};
+
+newtype pid_t as arraykey = int;
```

Comment:
Shall we expose `pid_t` as an `int`?

```suggestion
newtype pid_t as int = int;
```

Reply:
- do we need the `as` at all? why not leave them fully opaque?
- if we have `as int = int`, why not just `type pid_t = int` ?

Reply:
> do we need the as at all? why not leave them fully opaque?

At least `as arraykey` would be useful because some users might want to use pid_t as a key.

> if we have as int = int, why not just type pid_t = int ?
`newtype pid_t as int = int` would create a subtype of `int`, i.e. a `pid_t` is an `int`, but not all `int`s are `pid_t`s. Therefore we can let `waitpid` accepts a `pid_t` but not an arbitrary `int`.